hazelcast/src/main/java/com/hazelcast/nio/ConnectionManager.java

```diff
@@ -39,11 +39,32 @@
 
     void destroyConnection(Connection conn);
 
-    void shutdown();
-
+    /**
+     * Starts ConnectionManager, initializes its resources, starts threads etc.
+     * After start ConnectionManager becomes fully operational.
+     *
+     * If it's already started, then this method has no effect.
+     *
+     * @throws IllegalStateException if ConnectionManager is shutdown
+     */
     void start();
 
-    void restart();
+    /**
+     * Stops ConnectionManager, releases its resources, stops threads etc.
+     * When stopped ConnectionManager can be started again using {@link #start()}.
+     *
+     * This method has no effect if it's already stopped or shutdown.
```

Comment:
Can you add a few lines of comments what the purpose is of the stop. Apparently it is needed for the merging process.